src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -121,11 +122,14 @@ namespace cryptonote
       uint64_t    start_height;
       bool        prune;
       bool        no_miner_tx;
+      std::string client;
```

Comment:
That's an awful lot of structs that now get a new `client` field, and it all looks kind of un-elegant, but after brainstorming I was unable to come up with a alternative, better scheme: All that came to mind was either more complicated to implement or even threatened compatibility.

Why does `COMMAND_RPC_SUBMIT_RAW_TX::request_t` lack a `client` field?

I was musing that maybe it could be a good idea to add `client` to **all** structs, just out of principle and for symmetry. You could be sure then that no call "escaped" that should be able to carry a cost, and you could even introduce schemes like a very small "base cost" for bothering the daemon **at all** with an RPC call, even if only `get_height`.

Reply:
Why does client need to be specified in each RPC? Why don't we just associate a client at the beginning of a connection (via authentication) or after an `open_wallet` request and keep that associated to the session?

Reply:
SUBMIT_RAW_TX is not a RPC method. It's a mymonero RPC. It's been added there for reasons. I have a patch moving it to a mymonero specific header on github.

I suppose that now that we have SSL, we could cache the "I have checked this is an authorized client" in the connection structure and not recheck it every RPC. Sounds like something for later though.